src/arrays/multidimensional_arrays/hard/q1/MainTest.java

```diff
@@ -0,0 +1,112 @@
+package arrays.multidimensional_arrays.hard.q1;
+
+import global.BaseTest;
+import global.exceptions.InvalidClauseException;
+import global.tools.TestOption;
+import global.variables.Clause;
+import global.variables.clauses.IntegerLiteral;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.PlaceHolder;
+import global.variables.clauses.StringLiteral;
+import global.variables.wrappers.Optional;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MainTest extends BaseTest {
+    // Parsons
+
+    public Clause[] testSentence() {
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = "3 2 6 12 3 192 65 1";
+        return new Clause[]{
+                new StringLiteral("Enter the number of rows:"),
+                new Optional(new StringLiteral(" ")),
+                new NewLine(),
+                new StringLiteral("Enter the number of columns:"),
+                new Optional(new StringLiteral(" ")),
+                new NewLine(),
+                new PlaceHolder()
+        };
+    }
+
+    public void runMain() {
+        AscendDimensions.main(new String[0]);
+    }
+
+    static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of(0, 0),
+                Arguments.of(0, 1),
+                Arguments.of(1, 0),
+                Arguments.of(2, 3),
+                Arguments.of(4, 4),
+                Arguments.of(8, 11),
+                Arguments.of(56, 2),
+                Arguments.of(41, 13)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    void printsCorrectOutput(int rows, int columns) throws InvalidClauseException {
+        int[][] nums = arrayBuilder(rows, columns);
+        StringBuilder s = new StringBuilder(rows + " " + columns + " ");
+
+        for (int x = 0; x < rows; x++) {
+            for (int y = 0; y < columns; y++) {
+                s.append(nums[x][y]);
+                s.append(" ");
+            }
+        }
+
+        runWithInput(s.toString(), clauseBuilder(rows, columns));
+
+        for (int x = 0; x < rows; x++) {
+            for (int y = 0; y < columns; y++) {
+                assertEquals(nums[x][y], Integer.parseInt(getItemByName("row" + x + "col" + y)), "The number on row " + x + " column " + y + " is incorrect.");
```

Comment:
To follow out typical format, I think we should have the "Your program did not correctly ... " and then have the row and column identified in brackets " ... (the number on row x column y was incorrect.)